mysql-test/suite/innodb/t/innodb-page_compression_none.test

```diff
@@ -0,0 +1,30 @@
+--source include/have_innodb.inc
+--source include/have_sequence.inc
+
+--echo #
+--echo # MDEV-30825 innodb_compression_algorithm=0 (none) increments Innodb_num_pages_page_compression_error
+--echo #
+
+SET GLOBAL innodb_compression_algorithm=0, innodb_checksum_algorithm=strict_full_crc32, innodb_buffer_pool_size=8*1024*1024;
+SELECT VARIABLE_VALUE INTO @compress_errors FROM INFORMATION_SCHEMA.GLOBAL_STATUS WHERE VARIABLE_NAME = 'Innodb_num_pages_page_compression_error';
+
+
+CREATE TABLE t (c INT) page_compressed=1 page_compression_level=4 ENGINE=InnoDB;
+INSERT INTO t VALUES (1);
+SET GLOBAL innodb_checksum_algorithm=crc32;
+CREATE TABLE t1 (c INT) page_compressed=1 page_compression_level=4 ENGINE=InnoDB;
+INSERT INTO t1 VALUES (1);
+
+# do stuff to purge it out of innodb buffer pool - TODO how to do better.
+FLUSH TABLES;
+CREATE TEMPORARY TABLE b AS SELECT seq, REPEAT('aabbccddeeffgghhiijjkkllmmnn', 100000) FROM seq_1_to_1000;
+
+SELECT VARIABLE_VALUE - @compress_errors AS NUMBER_OF_ERRORS FROM INFORMATION_SCHEMA.GLOBAL_STATUS WHERE VARIABLE_NAME = 'Innodb_num_pages_page_compression_error';
+
+SET GLOBAL innodb_checksum_algorithm=DEFAULT, innodb_compression_algorithm=DEFAULT, innodb_buffer_pool_size=DEFAULT, innodb_checksum_algorithm=DEFAULT;
```

Comment:
I think that you should simply copy `mysql-test/suite/innodb/t/table_flags.combinations` to a `.combinations` file of this test, to have separate combinations for `full_crc32` and `crc32`.

For restoring global variables, I think that a better way is
```sql
SET @save_compression_algorithm=@@GLOBAL.innodb_compression_algorithm;
SET GLOBAL innodb_compression_algorithm=0;
--…
SET GLOBAL innodb_compression_algorithm=@save_compression_algorithm;
```
That would work even when the test is being invoked with a non-default parameter, such as
```sh
./mtr --mysqld=--loose-innodb-compression-algorithm=lz4
```